test/e2e/webdriver/driver.js

```diff
@@ -28,8 +29,31 @@ class Driver {
       // xpath locator.
       return By.xpath(locator.xpath);
     } else if (locator.text) {
-      // Providing a text prop, and optionally a tag, will use xpath to look
-      // for an element with the tag that has matching text.
+      // Providing a text prop, and optionally a tag or css prop, will use
+      // xpath to look for an element with the tag that has matching text.
+      if (locator.css) {
+        // When providing css prop we use cssToXPath to build a xpath string
```

Comment:
A majority of our usage of `until.elementMatchesText` was using a css selector then waiting until the returned element had matching text. Building an xpath combining CSS selection and text selection more closely mimics Playwright's API (page.waitForSelector('.hello >> text="hi"'). XPath IS slower than CSS selectors, BUT in the case of searching for text, it is not possible without XPath. Combining them into XPath is likely faster than issuing two statements looking for both a CSS selector AND then waiting for text to match.

In the end, though, this entire method (buildLocator) will go away and we will use the much more powerful and expressive playwright selector API.